src/test/java/com/google/sps/servlets/ProfessorDataTest.java

```diff
@@ -0,0 +1,196 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.data.ProfDataHolder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public final class ProfessorDataTest {
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private AddSchoolData schoolData;
+  private ProfessorData professorData;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    schoolData = new AddSchoolData();
+    professorData = new ProfessorData();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Mock HttpServletRequest request;
+  @Mock HttpServletRequest requestB;
+  @Mock HttpServletRequest requestC;
+
+  @Test
+  public void GettingRatingData_AllProperties() throws EntityNotFoundException {
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+
+    List<Object> expectedHoursListA = new ArrayList(Arrays.asList((long) 12));
+    List<Object> expectedDifficultyListA = new ArrayList(Arrays.asList((long) 7));
+    List<Object> expectedPerceptionListA = new ArrayList(Arrays.asList((double) 0.8));
+    List<Object> expectedCommentsListA = new ArrayList(Arrays.asList("Great"));
+
+    List<Object> expectedHoursListB = new ArrayList(Arrays.asList((long) 11));
+    List<Object> expectedDifficultyListB = new ArrayList(Arrays.asList((long) 6));
+    List<Object> expectedPerceptionListB = new ArrayList(Arrays.asList((double) 0.79));
+    List<Object> expectedCommentsListB = new ArrayList(Arrays.asList("Good"));
+
+    createRequest(
+        /* requestServelt */ request,
+        /* schoolName */ "MIT",
```

Comment:
Not super relevant now, but we'll probably need to lowercase this later as the email format is "mit.edu" (or do some kind of display name vs. email format kinda thing)

Reply:
^But something we should start thinking about, so implement if possible